qgsquick/from_qgis/quickgui/attributes/qgsquickattributecontroller.h

```diff
@@ -0,0 +1,184 @@
+/***************************************************************************
+ qgsquickattributecontroller.h
+  --------------------------------------
+  Date                 : 20.4.2021
+  Copyright            : (C) 2021 by Peter Petrik
+  Email                : zilolv@gmail.com
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef QGSQUICKATTRIBUTECONTROLLER_H
+#define QGSQUICKATTRIBUTECONTROLLER_H
+
+#include <QSortFilterProxyModel>
+#include <QObject>
+#include <QVariant>
+#include <memory>
+#include <QMap>
+#include <QVector>
+#include <QUuid>
+
+#include "qgis_quick.h"
+#include "qgsquickfeaturelayerpair.h"
+#include "qgsquickattributedata.h"
+
+#include "qgsfeature.h"
+#include "qgseditformconfig.h"
+#include "qgsexpressioncontext.h"
+#include "qgsattributeeditorcontainer.h"
+
+class QgsQuickAttributeFormModel;
+class QgsQuickAttributeFormProxyModel;
+class QgsQuickAttributeTabModel;
+class QgsQuickAttributeTabProxyModel;
+class QgsVectorLayer;
+class QgsQuickRememberAttributes;
+
+/**
+ * \ingroup quick
+ *
+ * This is implementation of the controller between QgsQuickAttribute*Model
+ * and the QML's QgsQuick.FeatureForm. It is responsible for creation of the
+ * models, and CRUD management between QgsFeature and the form. It connects
+ * the QgsFeature Fields on the backend side and the QML widgets on the user side
+ *
+ * Remember values functionality is done by QgsQuickRememberAttributes controller.
+ * Preview forms are not supported by this class.
```

Comment:
"Preview forms are not supported by this class."  -- I would leave this comment out

Reply:
ok